<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Miguel Tripp" />

<meta name="date" content="2020-11-22" />

<title>first-analysis</title>

<script src="site_libs/header-attrs-2.3/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RGhybrid</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/trippv/genomica">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">first-analysis</h1>
<h4 class="author">Miguel Tripp</h4>
<h4 class="date">2020-11-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-11-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>RGhybrid/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201122code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201122)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201122code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201122)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtrippvgenomicatree3b347426784aee3f40f385c0148bdaf7c4ccb7d1targetblank3b34742a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/trippv/genomica/tree/3b347426784aee3f40f385c0148bdaf7c4ccb7d1" target="_blank">3b34742</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtrippvgenomicatree3b347426784aee3f40f385c0148bdaf7c4ccb7d1targetblank3b34742a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/trippv/genomica/tree/3b347426784aee3f40f385c0148bdaf7c4ccb7d1" target="_blank">3b34742</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/Filtered_assembly_stats.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/first-analysis.Rmd</code>) and HTML (<code>docs/first-analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/genomica/5320a9e42589d83e3d261aa9a85196cf6c7a00ee/docs/first-analysis.html" target="_blank">5320a9e</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2020-11-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/trippv/genomica/blob/b80c56fc325c69e5d9ceba9f73098648b262656f/analysis/first-analysis.Rmd" target="_blank">b80c56f</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2020-11-22
</td>
<td>
Add my first analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Texto texto texto</p>
<p>In this document, we describe the steps for filtering the de novo assembly from pure red (<em>Haliotis rufescens</em>; RR) and hybrid abalone (<em>Haliotis rufescens</em> x <em>Haliotis fulgens</em>;RG).</p>
</div>
<div id="description-of-the-libraries" class="section level1">
<h1>Description of the libraries</h1>
<table>
<caption>Abalone libraries</caption>
<colgroup>
<col width="12%" />
<col width="23%" />
<col width="26%" />
<col width="37%" />
</colgroup>
<thead>
<tr class="header">
<th>Group</th>
<th>Sample</th>
<th>Library F</th>
<th>Library R</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>RAZ_UN</td>
<td>RAZ_UN_rep1</td>
<td>Trimmed_RAZ8_10_S43_L002_1P.fastq</td>
<td>Trimmed_RAZ8_10_S43_L002_2P.fastq</td>
</tr>
<tr class="even">
<td>RAZ_UN</td>
<td>RAZ_UN_rep2</td>
<td>Trimmed_RAZ11_13_S0_L009_1P.fastq</td>
<td>Trimmed_RAZ11_13_S0_L009_2P.fastq</td>
</tr>
<tr class="odd">
<td>RAZ_UN</td>
<td>RAZ_UN_rep3</td>
<td>Trimmed_RAZ14_16_S0_L009_1P.fastq</td>
<td>Trimmed_RAZ14_16_S0_L009_2P.fastq</td>
</tr>
<tr class="even">
<td>RAZ_M</td>
<td>RAZ_M_rep1</td>
<td>Trimmed_RAZ26_28_S46_L002_1P.fastq</td>
<td>Trimmed_RAZ26_28_S46_L002_2P.fastq</td>
</tr>
<tr class="odd">
<td>RAZ_M</td>
<td>RAZ_M_rep2</td>
<td>Trimmed_RAZ29_30_S47_L002_1P.fastq</td>
<td>Trimmed_RAZ29_30_S47_L002_2P.fastq</td>
</tr>
<tr class="even">
<td>RAZ_M</td>
<td>RAZ_M_rep3</td>
<td>Trimmed_RAZ31_32_S0_L009_1P.fastq</td>
<td>Trimmed_RAZ31_32_S0_L009_2P.fastq</td>
</tr>
<tr class="odd">
<td>RR_M</td>
<td>RR_M_rep1</td>
<td>Trimmed_RR51_53_S0_L009_1P.fastq</td>
<td>Trimmed_RR51_53_S0_L009_2P.fastq</td>
</tr>
<tr class="even">
<td>RR_M</td>
<td>RR_M_rep2</td>
<td>Trimmed_RR54_56_S50_L002_1P.fastq</td>
<td>Trimmed_RR54_56_S50_L002_2P.fastq</td>
</tr>
<tr class="odd">
<td>RR_M</td>
<td>RR_M_rep3</td>
<td>Trimmed_RR57_59_S51_L002_1P.fastq</td>
<td>Trimmed_RR57_59_S51_L002_2P.fastq</td>
</tr>
<tr class="even">
<td>RR_UN</td>
<td>RR_UN_rep1</td>
<td>Trimmed_RR87_89_S0_L009_1P.fastq</td>
<td>Trimmed_RR87_89_S0_L009_2P.fastq</td>
</tr>
<tr class="odd">
<td>RR_UN</td>
<td>RR_UN_rep2</td>
<td>Trimmed_RR90_92_S53_L002_1P.fastq</td>
<td>Trimmed_RR90_92_S53_L002_2P.fastq</td>
</tr>
<tr class="even">
<td>RR_UN</td>
<td>RR_UN_rep3</td>
<td>Trimmed_RR93_95_S0_L009_1P.fastq</td>
<td>Trimmed_RR93_95_S0_L009_2P.fastq</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div id="trinity-de-novo-assembly" class="section level1">
<h1>Trinity de novo assembly</h1>
<p>For the <em>de novo</em> assembly, we used Trinity v 2.4.0 including all RR and RG libraries. We used the dafult parameters except the <strong>no_bowtie</strong> option.</p>
</div>
<div id="filtering-step-for-the-assembly" class="section level1">
<h1>Filtering step for the assembly</h1>
<p>Filtering steps were as follows:</p>
<ol style="list-style-type: decimal">
<li>TPM filtering: Abundance estimations was made with RSEM using all libraries and the raw assembly. Therefore, a filtering step was applied to filter-out all transcripts with a TPM values &lt; 1.</li>
<li>CD-HIT: CD-HIT was used to cluster all contigs with an identity threshold of <strong>95%</strong></li>
</ol>
<p>The resulting assembly (CDHIT_TPM_filter) was compared with the following:</p>
<ul>
<li><strong>Raw_assembly</strong>: The initial assembly resulting from Trinity</li>
<li><strong>good_Trinity</strong>: The resulting contigs from running <em>TRANSRATE</em> in the Raw assebly.</li>
<li><strong>TPM_filter</strong>: First step of the filtering process. Contigs with TPM &lt; 1 were filtered out</li>
</ul>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div id="number-of-transcripts" class="section level2">
<h2>Number of transcripts</h2>
<p><img src="figure/first-analysis.Rmd/n_transcripts-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="number-of-trinity-genes" class="section level2">
<h2>Number of Trinity “genes”</h2>
<p><img src="figure/first-analysis.Rmd/n_genes-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="total-assembled-bases" class="section level2">
<h2>Total assembled bases</h2>
<p><img src="figure/first-analysis.Rmd/assembled_bases-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="uncovered-contigs" class="section level2">
<h2>uncovered contigs</h2>
<p>Is the he number of contigs that contain at least one base with no read coverage</p>
<p><img src="figure/first-analysis.Rmd/uncovered_contigs-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="transrate-score" class="section level2">
<h2>Transrate score</h2>
<p>The assembly score allows you to compare two or more assemblies made with the same reads. The score is designed so that an increased score is very likely to correspond to an assembly that is more biologically accurate<a href="https://hibberdlab.com/transrate/metrics.html">1</a>.</p>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-1-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="transrate-segmented-contigs" class="section level2">
<h2>transrate segmented contigs</h2>
<p>Is the number of contigs that have &gt;=50% estimated chance of being segmented</p>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-2-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="busco" class="section level2">
<h2>BUSCO</h2>
<div id="busco-results-for-meatzoa-database" class="section level3">
<h3>BUSCO results for Meatzoa database</h3>
<p>BUSCO uses sequence profiles embedded in lineage-specific datasets to assess the orthology status of predicted genes in the species under analysis. These consensus sequences are derived from Hidden Markov Model (HMM) profiles built from multiple sequence alignments of orthologs selected from OrthoDB and capture the conserved alignable amino acids across the species set, reducing any potential species bias that would result from pairwise alignments toward original sequences<a href="https://link.springer.com/protocol/10.1007/978-1-4939-9173-0_14">2</a>. BUSCO produces a report for each of the three modes of assessment using the same scoring scheme. Expected BUSCO genes can fall into different categories: C:complete [S:single-copy, D:duplicated], F:fragmented, and M:missing. These are reported as absolute numbers as well as percentage of the total BUSCO genes (n:) included in the dataset. To judge whether a score is satisfying, the user will have to consider the type of sequence first. A very good genome assembly should contain all BUSCO genes that were not lost during the evolution of the species, which cannot be precisely defined.</p>
<p>Non-model genome projects commonly report BUSCO scores ranging from 50% up to 95% complete, depending on the challenge posed by the species’ biology (e.g., genome size, amount of repetitive elements) and its taxonomic position.</p>
<blockquote>
<p>The duplication of a few BUSCO genes in a genome is compatible with a biological reality, as their evolution under single copy may be relaxed in some sublineages and the fact that we allowed duplications in up to 10% of the species when defining BUSCO markers [7]. However, a high duplication rate in a genome could denote a potential assembly of different haplotypes, a recent whole genome duplication [20], or technical artifacts that will have to be investigated. As mentioned earlier, the duplication rate of transcriptomes and annotated gene sets unfiltered for isoforms may be considerably higher. In some situation, the user will want to filter these out to decrease the duplication rate down to values expected in a genome. A high rate of fragmented BUSCO genes indicates issues in the sequencing and assembly process or the inability of the annotation pipeline to fully capture the complexity of some gene models. Turning fragmented BUSCO genes into complete is a good indicator of a significant improvement of the quality of an assembly, especially when supported by changes in other metrics such as N50.</p>
</blockquote>
<p><img src="figure/first-analysis.Rmd/Busco_MET-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="busco-results-for-mollusca-database" class="section level3">
<h3>BUSCO results for Mollusca database</h3>
<p><img src="figure/first-analysis.Rmd/Busco_MOL-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>Results from BUSCO demonstrate that keeping the “<em>good contings</em>” only, from using Transrate, results in a higher score but the number of BUSCOs is being reduced considerably. By using the TPM &gt; 1 and CD-HIT, the assembly score is also high but the number of missing BUSCOs is lower and the number of duplicated BUSCOs is reduced compared to the raw assembly.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19041)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] viridis_0.5.1     viridisLite_0.3.0 gplots_3.0.3      forcats_0.5.0    
 [5] stringr_1.4.0     dplyr_1.0.0       purrr_0.3.4       readr_1.3.1      
 [9] tidyr_1.1.0       tibble_3.0.1      ggplot2_3.3.1     tidyverse_1.3.0  
[13] workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6       lubridate_1.7.8    lattice_0.20-41    gtools_3.8.2      
 [5] assertthat_0.2.1   rprojroot_1.3-2    digest_0.6.25      R6_2.4.1          
 [9] cellranger_1.1.0   backports_1.1.7    reprex_0.3.0       evaluate_0.14     
[13] httr_1.4.1         pillar_1.4.4       rlang_0.4.6        readxl_1.3.1      
[17] rstudioapi_0.11    gdata_2.18.0       whisker_0.4        blob_1.2.1        
[21] rmarkdown_2.3      labeling_0.3       munsell_0.5.0      broom_0.5.6       
[25] compiler_4.0.0     httpuv_1.5.4       modelr_0.1.8       xfun_0.16         
[29] pkgconfig_2.0.3    htmltools_0.4.0    tidyselect_1.1.0   gridExtra_2.3     
[33] fansi_0.4.1        crayon_1.3.4       dbplyr_1.4.4       withr_2.2.0       
[37] later_1.0.0        bitops_1.0-6       grid_4.0.0         nlme_3.1-147      
[41] jsonlite_1.6.1     gtable_0.3.0       lifecycle_0.2.0    DBI_1.1.0         
[45] git2r_0.27.1       magrittr_1.5       scales_1.1.1       KernSmooth_2.23-16
[49] cli_2.0.2          stringi_1.4.6      farver_2.0.3       fs_1.4.1          
[53] promises_1.1.0     xml2_1.3.2         ellipsis_0.3.1     generics_0.0.2    
[57] vctrs_0.3.0        tools_4.0.0        glue_1.4.1         hms_0.5.3         
[61] yaml_2.2.1         colorspace_1.4-1   caTools_1.18.0     rvest_0.3.5       
[65] knitr_1.29         haven_2.3.1       </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
