---
title: "Differential_Expression"
author: "Miguel Tripp"
date: "2020-11-22"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE, fig.height = 8, fig.width = 12, fig.align = "center")
```


# Differential gene expression with **EdgeR**

The differential expression analysis was performed with EdgeR following the Trinity pipeline.

An exact test was used for the differential expression for each contrast. Transcritps with an FDR \< 0.01 and log2Fold change |\>| 1 were considered as deferentially expressed

```{r libraries, echo=FALSE, message=FALSE}
library(gplots)
library(RColorBrewer)
library(knitr)
library(patchwork)
library(tidyverse)
```

## MDS plot from the normalized data

TMM normalized data is used for this. This was also produced by the Trinity pipeline.

```{r MDSplot, echo = FALSE}
TMM_df <- read_delim("data/EdgeR/RSEM.isoform.TMM.EXPR.matrix.txt", delim = "\t")

TMM_df <- TMM_df %>%
  #filter Non-Eukaryote sequences
  #filter(X1 != x_good_nonEu$transcript_id) %>% #only one sequence in the DET list in NE
  column_to_rownames(var="X1") %>%
  #Change the name from RAZ (spanish) to RG
  rename_at(vars(starts_with("RAZ")), funs(str_replace(., "RAZ", "RG")))

#create a group variable
group = factor(sub("_rep.+", "", colnames(TMM_df)))

#create an DGE object for MDS plotting
library(edgeR)

dge <- DGEList(counts=TMM_df, group=group)

colors = c("purple","orange", "tomato", "salmon")[dge$samples$group]
names(colors) = levels(group)

MDS <- plotMDS(dge, col=colors, labels=colnames(dge$counts), plot = FALSE)

MDS_df <- data.frame(MDSX = MDS$x, MDSY = MDS$y, sample = names(MDS$x),
                     group = group)%>%
  mutate(sample = str_sub(sample,start = -4)) 

ggMDS <- ggplot(MDS_df, aes(x = MDSX, y = MDSY, color = group, fill = group ))+
  geom_point(size = 4)+
  scale_color_manual(values=c("purple", "orange", "tomato", "salmon"))+
  scale_fill_manual(values=c("purple", "orange", "tomato", "salmon"))+
    geom_polygon(alpha = 0.3)+
  geom_text(aes(label = sample),nudge_y = -0.05, nudge_x = -0.055)+
  annotate("text", x = c(-1, 0.40, 0.95, -0.50), y = c(0.75, 0.75, -0.35, -1.5), label = c("RR_M", "RR_UN", "RG_UN", "RG_M"), color = c("tomato", "salmon", "orange", "purple"))+
  theme_bw()+
  coord_cartesian(xlim = c(-2, 2))+
  theme(panel.grid = element_blank(),
        legend.position = "")+
  labs(x = "Leading logFC dim 1", y = "Leading logFC dim 2")

ggMDS

#ggsave(here("Results", "Res_DifferentialExpression", "MDSplot_DET_FrancescoData.pdf"), plot = ggMDS, dpi = 150, scale = 1.75)

```

## Number of differentially expressed transcripts

```{r DE_tables}

files <- list.files(path = "data/EdgeR",
                    full.names = T, pattern = "C1.DE.txt")

read.group.table <- function(file) {
  read.table(file, h = T, sep = "\t") %>%
    data.frame(., group = basename(file))
}

DE_df <- do.call(rbind, lapply(files, read.group.table)) #bind all tables in a single table

DE_df <- DE_df %>%
  
  #Remove the noneukaryote sequence
  #filter(transcript != x_good_nonEu$transcript_id) %>% 
  
  #change col names to correct Spanish labels
  rename_at(vars(starts_with("RAZ")), funs(str_replace(., "RAZ", "RG"))) %>%
  
  #change rows to correct spanglish
  mutate(sampleA = str_replace_all(sampleA, "RAZ", "RG")) %>% 
  mutate(sampleB = str_replace_all(sampleB, "RAZ", "RG")) %>% 
  
  #add new group column based on sampleA and sampleB columns
  mutate(group = paste0(sampleA, "-", sampleB)) %>% 
  
  #Add expression direction
  mutate(expression = ifelse(logFC > 0, "Upregulated", "Downregulated")) %>%
  mutate(expression = factor(expression, level = c("Upregulated", "Downregulated"))) %>%
  mutate(group = factor(group, level = c("RR_UN-RR_M", "RG_UN-RG_M",
                                         "RG_M-RR_M", "RG_UN-RR_UN")))

```

Number of DET per treatment:

```{r}

DET_barplot <- DE_df %>%
  dplyr::group_by(expression, group) %>%
  dplyr::count() %>%
ggplot(., aes(x = group, y = n, fill = expression)) +
  geom_bar(stat = "identity", position = position_dodge(0.9))+
  geom_text(aes(label = n), position = position_dodge(0.9), vjust = -0.5,
            size = 3)+
  theme_bw()+
  theme(panel.grid = element_blank(),
        legend.position=c(0.2,0.9),
        legend.background = element_blank())+
  labs(y = "Number of sequences", fill = "")

DET_barplot

```

## Heatmap of differentially expressed transcripts

```{r heatmap, echo = FALSE}

#Heatmap for all DE transcripts fro all contrasts


#vector with all DET across tested treatments
names_DET <- unique(DE_df$transcript)
length(names_DET)

TMM_matrix_DET <- TMM_df %>%
  tibble::rownames_to_column( "transcript") %>% 
  filter(transcript %in% names_DET)

data = TMM_matrix_DET[,-1]
sample_factoring = colnames(data)
data = as.matrix(data) # convert to matrix
data = t(scale(t(data), scale=T))
sample_dist = dist(t(data), method='euclidean')
hc_samples = hclust(sample_dist, method='complete')
gene_dist = dist(data, method='euclidean')
hc_genes = hclust(gene_dist, method='complete')
heatmap_data = data


#####
## Heatmap con con gplots
mypalette <- brewer.pal(11,"RdYlBu")
morecols <- colorRampPalette(mypalette)
# Set up colour vector for celltype variable
col.cell <- c("purple","orange", "tomato", "salmon")[dge$samples$group]


#heatmap
heatmap.2(heatmap_data,
          dendrogram='col', 
          Colv=as.dendrogram(hc_samples),
          col=rev(morecols(50)),
          main=" ",
          labRow = " ",
          srtRow = NULL,
          colRow = NULL,
          scale="none",
          density.info="none", trace="none", 
          ColSideColors=col.cell,
          margins = c(8,14),
          key=TRUE, keysize=1, 
          cexCol=1, key.title = NA,key.xlab ="log2(TMM)",
          lhei=c(1,3), lwid=c(1,5))


```
